.github/workflows/yocto-build-deploy.yml

```diff
@@ -257,44 +253,41 @@ jobs:
       # reference: https://github.com/balena-os/github-workflows/blob/master/.github/workflows/build_and_deploy.yml#L89
       # NOTE: This will not be necessary if we had a way to deploy artifacts and mark as final like with fleet releases
 
-      # Not needed as we should be able to get the tag from the caller workflow
-      # - name: 'Fetch latest tag'
-      #   id: get-latest-tag
-      #   if: ${{ inputs.check-merge-tests }}
-      #   uses: "actions-ecosystem/action-get-latest-tag@v1"
-
       # We're also checking out the tag in this step, so the subsequent build is done from the tagged version of the device repo
       - name: "Fetch merge commit"
         id: set-merge-commit
-        if: ${{ inputs.check-merge-tests }} # Left in the case of manual deploys where tests are failing but we had to force merge
+        if: ${{ github.event_name == 'push' }} # Left in the case of manual deploys where tests are failing but we had to force merge
```

Comment:
This comment needs updating, essentially we only want to check for previous runs on a tag event, otherwise we will use the boolean value for finalize, which is false by default.